### PR TITLE
cli: switch to Jest 29

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

The `@types/jest` packages is moved over to be a dependency controlled by the CLI rather than by each individual packages, since the CLI is what controls the Jest version.

Will target this for release in 1.7, because there are quite a few changes going out in 1.6 already that affect tests.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Looks like the e2e's are failing on windows only, some types stuff?

**Comment by Rugvip** [comment]
@mitchhentgesspotify huh, strange, haven't run into that here it seems. Luckily have a fix for that on the way over here though :grin: https://github.com/backstage/backstage/pull/12558/files#diff-d85fb4e7f1a6d317607efd5f614a608b237c8d5b7e5f8855c2e6364d417c4db0

**Review by freben — APPROVED** [review]
SWEET

File: packages/cli/config/jestYamlTransform.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const yaml = require('yaml');
+
+function createTransformer(config) {
+  const process = source => {
+    const json = JSON.stringify(yaml.parse(source), null, 2);
+    return { code: `module.exports = ${json}`, map: null };
+  };
+
+  const getCacheKey = sourceText => {
+    return createHash('md5')
```
Comment (mitchhentgesspotify) [inline packages/cli/config/jestYamlTransform.js:26]: Hey, sorry, I'm playing on your branch again: I think you want `crypto.createHash(...)` here :)

Comment (Rugvip) [reply packages/cli/config/jestYamlTransform.js:26]: Thanks x) Realized there's no point where we load any yaml in tests, so added a test for that too ._.

**Pull request merged**